default/AI/ResourcesAI.py

```diff
@@ -29,98 +38,258 @@
 lastFociCheck = [0]
 
 
-def _fill_data_dicts(planet_ids):
-    universe = fo.getUniverse()
-    newTargets.clear()
-    currentFocus.clear()
-    currentOutput.clear()
-    planets = [(pid, planetMap[pid]) for pid in planet_ids]
-    for pid, planet in planets:
-        currentFocus[pid] = planet.focus
-        currentOutput.setdefault(pid, {})[IFocus] = planet.currentMeterValue(fo.meterType.industry)
-        currentOutput[pid][RFocus] = planet.currentMeterValue(fo.meterType.research)
-        if IFocus in planet.availableFoci and planet.focus != IFocus:
-            fo.issueChangeFocusOrder(pid, IFocus)  # may not be able to take, but try
-    universe.updateMeterEstimates(planet_ids)
-    for pid, planet in planets:
-        itarget = planet.currentMeterValue(fo.meterType.targetIndustry)
-        rtarget = planet.currentMeterValue(fo.meterType.targetResearch)
-        if planet.focus == IFocus:
-            newTargets.setdefault(pid, {})[IFocus] = (itarget, rtarget)
-            newTargets.setdefault(pid, {})[GFocus] = [0, rtarget]
-        else:
-            newTargets.setdefault(pid, {})[IFocus] = (0, 0)
-            newTargets.setdefault(pid, {})[GFocus] = [0, 0]
-        # if currentFocus[pid] == MFocus:
-        # newTargets[pid][MFocus] = (mtarget, rtarget)
-        if RFocus in planet.availableFoci and planet.focus != RFocus:
-            fo.issueChangeFocusOrder(pid, RFocus)  # may not be able to take, but try
-    universe.updateMeterEstimates(planet_ids)
-    for pid, planet in planets:
-        can_focus = planet.currentMeterValue(fo.meterType.targetPopulation) > 0
+class PlanetFocusInfo(object):
+    """ The current, possible and future foci and output of one planet."""
+    def __init__(self, planet):
+        self.planet = planet
```

Comment:
I used planet instead of pid to avoid the cost of a `fo.getUniverse()` call in every PlanetFocusInfo every game turn.  This way fo.getUniverse() is called once per turn in PlanetFocusManager().

I didn't profile it, so it is probably premature optimization.
Changing it is trivial. 

Should I change it?